src/box/sql/vdbe.c

```diff
@@ -1976,13 +1976,26 @@ case OP_Column: {
  * P2 fields, then extract a NULL.
  */
 case OP_Fetch: {
-	struct vdbe_field_ref *field_ref =
-		(struct vdbe_field_ref *) p->aMem[pOp->p1].u.p;
-	uint32_t field_idx = pOp->p2;
-	struct Mem *dest_mem = vdbe_prepare_null_out(p, pOp->p3);
-	if (vdbe_field_ref_fetch(field_ref, field_idx, dest_mem) != 0)
+	struct vdbe_field_ref *ref = p->aMem[pOp->p1].u.p;
+	struct tuple_format *format = ref->format;
+	size_t id;
+	if (pOp->p4type != P4_DYNAMIC) {
+		id = pOp->p2;
+	} else {
+		const char *name = pOp->p4.z;
+		for (id = 0; id < format->dict->name_count; id++) {
+			if (strcmp(format->dict->names[id], name) == 0)
+				break;
+		}
+		if (id == format->dict->name_count) {
+			diag_set(ClientError, ER_SQL_CANT_RESOLVE_FIELD, name);
+			goto abort_due_to_error;
+		}
```

Comment:
You can also use `tuple_fieldno_by_name()`. To get field name hash you can use the global function `field_name_hash()`.

Reply:
Thanks, fixed.